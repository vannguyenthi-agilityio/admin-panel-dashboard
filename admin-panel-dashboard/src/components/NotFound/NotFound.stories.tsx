import type { Meta, StoryObj } from '@storybook/react';

// Components
import NotFound from './';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta : Meta<typeof NotFound> = {
  title: 'Components/Header',
  component: NotFound,
  tags: ['autodocs'],
  argTypes: {
    className: { description: "Class name taiwind to style for text page, type is string" },
    message: { description: "Message content to show on page, type is string" },
    size: { description: "Size text, padding to style on page, type is SIZE_TYPE: xs, sm, md, lg" }
  },
};

export default meta;

type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
  args: {},
};
