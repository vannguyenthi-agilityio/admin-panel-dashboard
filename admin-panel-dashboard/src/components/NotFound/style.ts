import { cva } from 'class-variance-authority';

export const notFoundStyles = cva(
  'flex items-center justify-center w-full min-h-[calc(100vh-200px)]',
  {
    variants: {
      variant: {
        primary: ["bg-transparent", "text-red-500"],
        secondary: ["bg-gray-100", "text-darker"],
      },
      size: {
        xs: ["text-xs", "px-2", "py-0"],
        sm: ["text-sm", "px-2", "py-2"],
        md: ["text-md", "px-2", "py-2"],
        lg: ["text-lg", "px-2", "py-2"],
      },
    },
    defaultVariants: {
      variant: "primary",
      size: "md",
    },
  }
);
