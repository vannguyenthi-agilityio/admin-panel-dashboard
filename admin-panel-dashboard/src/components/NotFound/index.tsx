// Helpers
import { clsxMerge } from '@/helpers';

// Types
import { SIZE_TYPE } from '@/types'

import { notFoundStyles } from './style';

interface INotFound {
  message?: string;
  className?: string;
  size?: SIZE_TYPE;
}

const NotFound = ({ message="Not Found", className, size }: INotFound) => (
  <div
    className={clsxMerge(
      notFoundStyles({ 
        size,
      }),
      className
    )}
  >
    <p className="font-bold">{message.toString()}</p>
  </div>
);

export default NotFound;
