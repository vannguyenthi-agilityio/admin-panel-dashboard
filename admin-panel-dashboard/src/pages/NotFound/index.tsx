// Components
import IBreadcrumb from '@/components/commons/Breadcrumb';
import { NotFound } from '@/components';

const NotFoundPage = () => {

  return (
    <div className='min-h-screen mx-auto pt-[100px] px-6'>
      <h4 className="text-[20px] md:text-lg text-darker font-medium">Not Found</h4>
      <IBreadcrumb />
      <NotFound message="Page Not Found"/>
    </div>
  );
};

export default NotFoundPage;
