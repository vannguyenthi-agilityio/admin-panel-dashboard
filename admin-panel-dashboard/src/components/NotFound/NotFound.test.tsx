import {
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import NotFound from '.'

describe('NotFound component', () => {
  it('Renders NotFound default component', () => {
    const {container} = render(<NotFound />)
    expect(container).toMatchSnapshot();
  });
})
