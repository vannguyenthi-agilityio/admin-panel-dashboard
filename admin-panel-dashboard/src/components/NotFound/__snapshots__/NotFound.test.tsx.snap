// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component Renders NotFound default component 1`] = `
<div>
  <div
    class="flex items-center justify-center w-full min-h-[calc(100vh-200px)] bg-transparent text-red-500 text-md px-2 py-2"
  >
    <p
      class="font-bold"
    >
      Not Found
    </p>
  </div>
</div>
`;
